app/code/Dev/ProductComment/Block/Widget/Posts.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace Dev\ProductComment\Block\Widget;
+
+use Magento\Catalog\Api\ProductRepositoryInterface;
+use Magento\Framework\Api\SearchCriteriaBuilder;
+use Magento\Framework\View\Element\Template;
+use Magento\Widget\Block\BlockInterface;
+use \Magento\Catalog\Helper\Image;
+use Magento\Framework\Pricing\Helper\Data;
+
+class Posts extends Template implements BlockInterface
+{
+
+    protected $_template = "widget/posts.phtml";
+    /**
+     * @var ProductRepositoryInterface
+     */
+    private $productRepository;
+    /**
+     * @var SearchCriteriaBuilder
+     */
+    private $criteriaBuilder;
+
+    /**
+     * Posts constructor.
+     * @param Template\Context $context
+     * @param ProductRepositoryInterface $productRepository
+     * @param SearchCriteriaBuilder $criteriaBuilder
+     */
+
+    private $imageHelper;
+    /**
+     * @var \Magento\Directory\Model\Currency
+     */
+    private $currency;
+
+    /**
+     * Posts constructor.
+     * @param Template\Context $context
+     * @param ProductRepositoryInterface $productRepository
+     * @param SearchCriteriaBuilder $criteriaBuilder
+     * @param Image $imageHelper
+     * @param \Magento\Directory\Model\Currency $currency
+     * @param array $data
+     */
+    public function __construct(
+        Template\Context $context,
+        ProductRepositoryInterface $productRepository,
+        SearchCriteriaBuilder $criteriaBuilder,
+        Image $imageHelper,
+        \Magento\Directory\Model\Currency $currency,
+        array $data = [])
+    {
+        parent::__construct($context, $data);
+        $this->productRepository = $productRepository;
+        $this->criteriaBuilder = $criteriaBuilder;
+        $this->imageHelper = $imageHelper;
+        $this->currency = $currency;
+    }
+
+
+    public function getProductCollection($maxProducts)
+    {
+        $criteria = $this->criteriaBuilder
+            ->addFilter('ProductComment', 'yes')
+            ->create()
+            ->setPageSize($maxProducts);
+
+        return $this->productRepository
+            ->getList($criteria)
+            ->getItems();
+    }
+
+
+    public function getItemImage($productId)
+    {
+        try {
+            $_product = $this->productRepository->getById($productId);
```

Comment:
You already have a product object from the collection. Could use it instead of loading the product again.